central/sensorupgrade/service/service_impl_test.go

```diff
@@ -44,48 +50,68 @@ func (s *SensorUpgradeServiceTestSuite) Test_UpdateSensorUpgradeConfig() {
 	testCases := map[string]struct {
 		req               *v1.UpdateSensorUpgradeConfigRequest
 		managedCentral    bool
+		upgraderDisabled  bool
 		expectedErr       error
 		upsertTimesCalled int
 	}{
-		"Error: No config": {
+		"Nil config should yield an error": {
 			req:               &v1.UpdateSensorUpgradeConfigRequest{Config: nil},
 			expectedErr:       errox.InvalidArgs,
 			upsertTimesCalled: 0,
 		},
-		"Error: can't set toggle = true on managed centrals": {
+		"Enabling upgrader through the config should work on managed centrals": {
 			managedCentral: true,
 			req: &v1.UpdateSensorUpgradeConfigRequest{
 				Config: configWith(true),
 			},
+			expectedErr:       nil,
+			upsertTimesCalled: 1,
+		},
+		"Enabling upgrader through the config should fail on managed centrals if upgrader is explicitly disabled": {
+			managedCentral:   true,
+			upgraderDisabled: true,
+			req: &v1.UpdateSensorUpgradeConfigRequest{
+				Config: configWith(true),
+			},
 			expectedErr:       errox.InvalidArgs,
 			upsertTimesCalled: 0,
 		},
-		"Success: can set toggle = false on managed centrals": {
+		"Disabling upgrader through the config should work on managed centrals": {
 			managedCentral: true,
 			req: &v1.UpdateSensorUpgradeConfigRequest{
 				Config: configWith(false),
 			},
 			upsertTimesCalled: 1,
 		},
-		"Success: can set toggle = true on non-managed centrals": {
+		"Enabling upgrader through the config should work on non-CS centrals": {
 			managedCentral: false,
 			req: &v1.UpdateSensorUpgradeConfigRequest{
 				Config: configWith(true),
 			},
 			upsertTimesCalled: 1,
 		},
-		"Success: can set toggle = false on non-managed centrals": {
+		"Disabling upgrader through the config should work on non-CS centrals": {
 			managedCentral: false,
 			req: &v1.UpdateSensorUpgradeConfigRequest{
 				Config: configWith(false),
 			},
 			upsertTimesCalled: 1,
 		},
+		"Disabling upgrader through the config should not yield an error on non-CS centrals even if upgrader is explicitly disabled": {
+			managedCentral:   false,
+			upgraderDisabled: true,
+			req: &v1.UpdateSensorUpgradeConfigRequest{
+				Config: configWith(false),
+			},
+			expectedErr:       nil,
+			upsertTimesCalled: 1,
+		},
 	}
 
 	for caseName, testCase := range testCases {
 		s.Run(caseName, func() {
 			s.T().Setenv(env.ManagedCentral.EnvVar(), strconv.FormatBool(testCase.managedCentral))
+			s.T().Setenv(env.SensorUpgraderEnabled.EnvVar(), strconv.FormatBool(!testCase.upgraderDisabled))
```

Comment:
Nit: If we have a flag for `StuffEnabled`, it feels more logical to have a bool `thingEnabled` as well, instead of `!thingDisabled`. That might be more of a personal preference though.

Reply:
Yeah, I wanted to keep the diff minimum, but that does not make sense. I will apply the suggestion

Reply:
Having the default value set to false was more compact, but now it should be better readable: [c615fc9](https://github.com/stackrox/stackrox/pull/12150/commits/c615fc961db8dc9a83cd22f8b129af92118694ac)